### PR TITLE
[Fixes #7068] mixed case TopicCategories are mangled during layer upload

### LINKED ISSUE #7068: mixed case TopicCategories are mangled during layer upload
## Expected Behavior
Uploading a shapefile with `<gmd:MD_TopicCategoryCode>utilitiesCommunication</gmd:MD_TopicCategoryCode>` or `gmd:MD_TopicCategoryCode>plannincCadastre</gmd:MD_TopicCategoryCode>` should show up in the predefined categories.

## Actual Behavior
A new Catageory "utilitiesCommunication" resp. "plannincCadastre" without an Icon is created. The predefined categories "Utilities Communication" or "Planning Cadastre" are not used.

## Steps to Reproduce the Problem

  1. Upload a shapefile with an ISO-XML containing one of the above mentioned MD_TopicCategoryCode codes
  2. View the layer or the layers metadata

## Specifications

  - GeoNode version: master
  - Installation method (manual, GeoNode Docker, SPCGeoNode Docker): manual
  - Platform: Linux X64

### PR DESCRIPTION
**Pull request by stefmec**
Categories are stored in mixed case in the database but where converted to lowercase during upload. This fix keeps the case.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/GeoNode/geonode/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in the documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for contrib modules):
- [x] There is a ticket in https://github.com/GeoNode/geonode/issues describing the issue/improvement/feature (a notable exemption is, changes not visible to end-users)
- [x] The issue connected to the PR must have Labels and Milestone assigned
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[Fixes #<issue_number>] Title of the Issue"
- [x] New unit tests have been added covering the changes, unless there is an explanation on why the tests are not necessary/implemented
- [ ] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the QA checks: flake8 geonode
- [ ] Commits changing the **settings**, **UI**, **existing user workflows**, or adding **new functionality**, need to include documentation updates
- [ ] Commits adding **new texts** do use gettext and have updated .po / .mo files (without location infos)

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

### CODE REVIEW

File: geonode/layers/tests.py [hunk 1]
```diff
@@ -1342,6 +1342,65 @@ def setUp(self):
...
+    @on_ogc_backend(geoserver.BACKEND_PACKAGE)
+    @override_settings(UPLOADER=GEONODE_REST_UPLOADER)
+    def test_geonode_upload_kml_and_import_isocategory(self):
+        """
+        Ensure a KML-File can be uploaded and the category is imported correctly from accompanying ISO-XML
+        """
+
+        filename_suffix_list = [
+        'structure',        # lower case default category
+        'planningCadastre', # mixed case default category
```
Comment (gannebamm) [inline geonode/layers/tests.py:1354]: @stefmec please fix the PEP8 formatting errors:

^@^@geonode/layers/tests.py:1354:28: E261 at least two spaces before inline comment
geonode/layers/tests.py:1368:21: E128 continuation line under-indented for visual indent
geonode/layers/tests.py:1370:21: E128 continuation line under-indented for visual indent
geonode/layers/tests.py:1404:5: E303 too many blank lines (2

File: geonode/layers/tests.py [hunk 2]
```diff
@@ -1382,10 +1442,12 @@ def test_geonode_rest_layer_uploader(self):
             if isinstance(content, bytes):
                 content = content.decode('UTF-8')
             data = json.loads(content)
+            logger.info(data)
             # Check success
             self.assertTrue(data['success'])
             _lname = data['url'].split(':')[-1]
             _l = Layer.objects.get(name=_lname)
+            print(_l.name)
```
Comment (afabiani) [inline geonode/layers/tests.py:1450]: Do not use print statements please. Use logger instead

**Review by afabiani — CHANGES_REQUESTED** [review]

**Review by afabiani — APPROVED** [review]

**Comment by gannebamm** [comment]
The uuid test fails with an http error? I do not think that error is related to the changes in this PR. Has circleCI timeout issues?

**Comment by t-book** [comment]
@gannebamm This is not connected. Unfortunately, the Upload-Test is failing regularly. @afabiani and I already tried some things to solve it. No luck so far. Maybe https://circleci.com/docs/2.0/ssh-access-jobs/ could shed light into ...

**Comment by gannebamm** [comment]
The backports failed? @afabiani do you know why? Shouldn't be 3.2.x specific Code in this PR.

**Comment by afabiani** [comment]
@gannebamm backporter on both 3.2.x and 3.1.x

**Pull request merged**